plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/_impl.py

```diff
@@ -157,11 +159,32 @@ def sweep(self, arguments: List[str]) -> None:
             )
             self.validate_batch_is_legal(overrides)
             returns = self.launcher.launch(overrides, initial_job_idx=self.job_idx)
-            self.job_idx += len(returns)
             # would have been nice to avoid waiting for all jobs to finish
             # aka batch size Vs steady state (launching a new job whenever one is done)
+            self.job_idx += len(returns)
+            # check job status and prepare losses
+            failures = 0
+            to_tell: List[Tuple[ng.p.Parameter, float]] = []
             for cand, ret in zip(candidates, returns):
-                loss = direction * ret.return_value
+                if ret.status != utils.JobStatus.COMPLETED:
+                    to_tell.append((cand, float("inf")))
+                    failures += 1
+                    try:
+                        ret.return_value
+                    except HydraJobException as e:
+                        log.info(f"Returning infinity for failed experiment: {e}")
+                else:
+                    to_tell.append((cand, direction * ret.return_value))
+            # raise if too many failures
+            if failures / len(returns) > self.opt_config.max_failure_rate:
+                log.error(
+                    f"Failed {failures} times out of {len(returns)} "
+                    f"with max_failure_rate={self.opt_config.max_failure_rate}"
+                )
+                for ret in returns:
+                    ret.return_value  # delegate raising to JobReturn, with actual traceback
```

Comment:
What does the traceback look like here?

Reply:
Something like this:
<img width="611" alt="Screen Shot 2021-04-13 at 1 30 36 PM" src="https://user-images.githubusercontent.com/11429864/114545869-c6bafe80-9c5c-11eb-83d8-0ac1d5cf727e.png">